src/components/eip4626/user/VaultWithdraw.tsx

```diff
@@ -73,100 +73,141 @@ export function VaultWithdraw({ vaultAddress }: VaultInfoProps) {
     });
   };
 
-  //@TODO call not working for small amounts with decimals
-  // const { data: previewWithdrawData } = useReadHederaVaultPreviewWithdraw(
-  //   vaultAddress,
-  //   form.values.amount,
-  // );
+  const maxAmount = useMemo(
+    () => Number(formatBalance(shareUserBalance)),
+    [shareUserBalance],
+  );
+
+  const handleUpdateAmountWithPercent = (percentage: number | string) => {
+    const calculatedAmount =
+      typeof percentage === "number"
+        ? (maxAmount * percentage) / 100
+        : maxAmount;
+
+    if (calculatedAmount) {
+      withdrawForm.setFieldValue("amount", calculatedAmount);
+
+      if (calculatedAmount <= maxAmount) {
+        setMaxAmountError(false);
+      } else {
+        setMaxAmountError(true);
+      }
+    }
+  };
 
-  const { data: userContribution } =
-    useReadHederaVaultUserContribution(vaultAddress);
+  const [maxAmountError, setMaxAmountError] = useState(false);
 
   return (
     <>
-      <Heading size={"sm"}>Withdraw asset from vault</Heading>
-
-      <form onSubmit={form.handleSubmit}>
-        <VStack gap={2} alignItems="flex-start">
-          <FormControl isRequired>
-            <FormLabel>Amount of asset to withdraw</FormLabel>
-            <NumberInput
-              name="amount"
-              variant="outline"
-              value={form.values.amount}
-              clampValueOnBlur={false}
-              onChange={(val) => form.setFieldValue("amount", val)}
+      <form onSubmit={withdrawForm.handleSubmit}>
+        <Flex direction="column" gap="4">
+          <Flex
+            direction="row"
+            width="100%"
+            pt="2"
+            pb="2"
+            justifyContent="space-between"
+          >
+            <Text fontWeight="800" fontSize={14}>
+              Withdraw Amount{": "}
+              {withdrawForm.values?.amount}
+            </Text>
+            <Text fontWeight="600" fontSize={14}>
+              Max: {maxAmount}
+            </Text>
+          </Flex>
+          <Flex width="100%" justifyContent="space-between" gap="1">
+            <Input
+              value={withdrawForm.values?.amount}
+              onChange={(e) => {
+                const calculatedAmount = new BigNumber(
+                  e.target.value,
+                ).toNumber();
```

Comment:
are we sure we want to convert to JS number here with `.toNumber();`? i feel we might have issues with operating actual big numbers, converting to string might be safer, wdyt?

Reply:
> are we sure we want to convert to JS number here with `.toNumber();`? i feel we might have issues with operating actual big numbers, converting to string might be safer, wdyt?

Hm, no, i don't think it should be as an issue here. + we want to perform some operations like: `calculatedAmount <= maxAmount` and those operations actually performs only with numbers.